gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -200,6 +201,9 @@ class YarnService extends AbstractIdleService {
   private volatile boolean shutdownInProgress = false;
 
   private final boolean jarCacheEnabled;
+  private final WorkerProfile baselineWorkerProfile;
+  private final AtomicLong allocationRequestIdGenerator = new AtomicLong(0L);
+  private final ConcurrentMap<Long, WorkerProfile> allocationRequestIdToWorkerProfile = new ConcurrentHashMap<>();
```

Comment:
this name works and is ok... but, considering two alternatives:
```
WorkerProfile workerProfile = this.allocationRequestIdToWorkerProfile.get(allocationRequestId);
WorkerProfile workerProfile = this.workerProfileByAllocationRequestId.get(allocationRequestId);
```
I prefer the second for its focus on the fact that the worker profile (that we're here to retrieve) is the more important half of the mapping.